src/modules/network.cpp

```diff
@@ -13,7 +13,24 @@ namespace modules {
   network_module::network_module(const bar_settings& bar, string name_)
       : timer_module<network_module>(bar, move(name_)) {
     // Load configuration values
+    // m_interface could be a regex as well
     m_interface = m_conf.get(name(), "interface", m_interface);
+    std::regex interface_regex(m_interface);
+
+    struct ifaddrs* ifaddrs;
+    getifaddrs(&ifaddrs);
```

Comment:
This leaks memory. You can also detect that yourself if you configure polybar using the `SANITIZE` build type: `cmake -DCMAKE_BUILD_TYPE=Sanitize ..`